src/components/tooltip/Tooltip.spec.tsx

```diff
@@ -0,0 +1,292 @@
+import * as React from 'react';
+import {
+  screen,
+  render,
+  fireEvent,
+  waitForElementToBeRemoved,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Tooltip from './Tooltip';
+import Button from '../buttons/Button';
+import Icon from '../icons/Icon';
+
+export function RenderTooltip(props: {
+  defaultOpen?: boolean;
+  children: React.ReactNode;
+}) {
+  return (
+    <Tooltip defaultOpen={props.defaultOpen}>
+      <Tooltip.Element label="Copy to clipboard" />
+      <Tooltip.Trigger>{props.children}</Tooltip.Trigger>
+    </Tooltip>
+  );
+}
+
+// JSDom (which is used by jest) does not implement layout/rendering.
+// we create this mock to simply simulate a desktop view with a width of 1000
+function useResizeObserverMock() {
```

Comment:
let's keep "use*" prefix in naming strictly for hooks.

Reply:
This is a mock of `useResizeObserver`, so use* is fine in this case.

Have you tried to mock ResizeObserver instead? It would be more flexible and we can have full control over it.

Something like this:

```suggestion
global.ResizeObserver = jest.fn().mockImplementation(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
}))
```

Reply:
true, missed the part it's a mock for a hook, agree that it can be mocked deeper.